cgra/CgraRTL.py

```diff
@@ -105,13 +105,17 @@ def construct(s, DataType, PredicateType, CtrlPktType, CtrlSignalType,
     s.recv_from_cpu_pkt //= s.controller.recv_from_cpu_pkt
 
     # Connects ring with each control memory.
-    for i in range(s.num_tiles):
-      s.ctrl_ring.send[i] //= s.tile[i].recv_ctrl_pkt
+    s.ctrl_ring.send[0] //= s.controller.recv_from_ctrl_ring_ctrl_pkt
+    for i in range(1, s.num_tiles + 1):

```

Comment:
Once `s.ctrl_ring.send[s.num_tiles] //= s.controller.recv_from_ctrl_ring_ctrl_pkt`, we can have `for i in range(s.num_tiles):` here.